subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1370,6 +1426,13 @@ public DeprecatableConfiguration deprecateForResolution(String... alternativesFo
         return this;
     }
 
+    @Override
+    public Configuration resolveConsistentlyWith(Configuration versionsSource) {
+        this.consistentResolutionSource = (ConfigurationInternal) versionsSource;
+        this.consistentResolutionReason = "version resolved in " + versionsSource + " by consistent resolution";
```

Comment:
I think this can just be a local variable in `registerConsistentResolutionConstraint`, no need to hold onto derived state in this class (especially as it's never reused outside the register call within this particular configuration)

Reply:
It can but then we would create duplicate strings. It's a performance optimization in order to reduce the memory footprint, which can be quite a lot in dependency management, by experience.